atom/browser/native_window_mac.mm

```diff
@@ -438,22 +451,17 @@ - (void)drawRect:(NSRect)dirtyRect {
   if (closable) {
     styleMask |= NSClosableWindowMask;
   }
+  if ((titleBarStyle == "hidden") || (titleBarStyle == "hidden-inset")) {
+    // The window without titlebar is treated the same with frameless window.
+    set_has_frame(false);
```

Comment:
this also prevents the title from being set in `NativeWindowMac::SetTitle` which drops these windows out of the window menu